source/wasm/spirv-tools.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "spirv-tools/libspirv.hpp"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+using namespace emscripten;
+
+auto print_msg_to_stderr = [](spv_message_level_t, const char*,
```

Comment:
You should be able to declare this as a regular function.
The constructor for the  std::function will auto-convert it, I think.

Also, the style guide recommends not creating static-duration objects with non-trivial destructors.  https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables 
 I think the lambda falls into that category.

Reply:
I think I got this from the example: https://github.com/KhronosGroup/SPIRV-Tools/blob/master/examples/cpp-interface/main.cpp#L43, Anyway will take a look.

Reply:
Guess I had just blindly copy/pasted from the example - regular function works just fine here.